### PR TITLE
Harden velocypack validator after fuzz testing

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Backport of https://github.com/arangodb/arangodb/pull/15649

Harden velocypack validator after fuzz testing.
Changes are from upstream https://github.com/arangodb/velocypack/pull/111

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [x] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [x] Backports
  - [x] Backport for 3.8: this PR
  - [x] Backport for 3.7: https://github.com/arangodb/arangodb/pull/15663

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

**Comment by joerg84** [comment]
@jsteemann Is this one ready for review?

**Review by mchacki — APPROVED** [review]
LGTM.
Same comments as for 3.9 variant apply.
However as I do not consider any of my comments "critical" by any means, please merge this as is please to avoid unnecessary additions to stable branch

File: 3rdParty/velocypack/src/Parser.cpp [hunk 1]
```diff
@@ -118,6 +118,17 @@ int Parser::skipWhiteSpace(char const* err) {
   throw Exception(Exception::ParseError, err);
 }
 
+void Parser::increaseNesting() { 
+  if (++_nesting >= options->nestingLimit) {
+    throw Exception(Exception::TooDeepNesting);
+  }
```
Comment (mpoeter) [inline 3rdParty/velocypack/src/Parser.cpp:124]: Here we modify the local state before throwing the exception. AFAIK the parser object can be reused, but I don't think this local state is reinitialized correctly.

**Review by mpoeter — APPROVED** [review]
LGTM, but can we add some unit tests for the other fixes as well, i.e., too deep nesting and incorrect index tables?

File: 3rdParty/velocypack/src/Parser.cpp (hunk 1 above)
Comment (jsteemann) [reply 3rdParty/velocypack/src/Parser.cpp:124]: In devel, the `_nesting` attribute of the Parser is actually reset in the `parse()` method correctly, so this is not an issue there.
However, this change was missing in the backports. 
Good catch!
VelocyPack test case added via https://github.com/arangodb/velocypack/pull/122

**Review by mpoeter — APPROVED** [review]

**Pull request merged**